docs/reference/method/MongoDBClient-addSubscriber.txt

```diff
@@ -0,0 +1,130 @@
+-----------------
+
+.. include:: /includes/extracts/error-invalidargumentexception.rst
+
+Example
+-------
+
+Create a :phpclass:`MongoDB\\Driver\\Monitoring\\CommandSubscriber` that
+logs all events:
+
+.. code-block:: php
+
+    <?php
+
+        use MongoDB\Driver\Monitoring\CommandSubscriber;
+        use MongoDB\Driver\Monitoring\CommandStartedEvent;
+        use MongoDB\Driver\Monitoring\CommandSucceededEvent;
+        use MongoDB\Driver\Monitoring\CommandFailedEvent;
+
+        class LogCommandSubscriber implements CommandSubscriber
+        {
+            private $stream;
+            public function __construct($stream)
+            {
+                $this->stream = $stream;
+            }
+
+            public function commandStarted(CommandStartedEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Started command #%d "%s": %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    Document::fromPHP($event->getCommand())->toCanonicalExtendedJSON(),
+                    PHP_EOL,
+                ));
+            }
+
+            public function commandSucceeded(CommandSucceededEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Succeeded command #%d "%s" in %d microseconds: %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    $event->getDurationMicros(),
+                    json_encode($event->getReply()),
+                    PHP_EOL,
+                ));
+            }
+
+            public function commandFailed(CommandFailedEvent $event): void
+            {
+                fwrite($this->stream, sprintf(
+                    'Failed command #%d "%s" in %d microseconds: %s%s',
+                    $event->getRequestId(),
+                    $event->getCommandName(),
+                    $event->getDurationMicros(),
+                    $event->getError()->getMessage(),
+                    PHP_EOL,
+                ));
+            }
+        }
+
+The subscriber can then be registered with a Client:
+
+.. code-block:: php
+
+    <?php
+
+        $client = new MongoDB\Client();
+        $subscriber = new LogCommandSubscriber(STDERR);
+
+        $client->addSubscriber($subscriber);
+
+        $client->test->users->insertOne(['username' => 'alice']);
+
+The above code will log the following to err output:
+
+.. code-block:: text
+
+    Started command #1 "insert": { "insert" : "users", "ordered" : true, "$db" : "test", "lsid" : { "id" : { "$binary" : { "base64" : "dKTBhZD7Qvi0vUhvR58mCA==", "subType" : "04" } } }, "documents" : [ { "username" : "alice", "_id" : { "$oid" : "655d1fca12e81018340a4fc2" } } ] }
+    Succeeded command #1 "insert" in 3349 microseconds: {"n":1,"ok":1}
+
+See Also
+--------
+
+- :phpmethod:`MongoDB\\Client::removeSubscriber()`
+- :php:`Application Performance Monitoring (APM)
```

Comment:
No objection to make these lines longer for readability. There is prior art for that, such as `MongoDBClient-getWriteConcern.txt`.